### PR TITLE
Improvements to Boxplot dialog

### LINKED ISSUE #8995: Small puzzle with boxplot dialog?
@jkmusyoka and @MeSophie as I write this issue, I have a suggestion for solving the puzzle.  It is a seasonal plot for Dodoma monthly rainfall totals.  The data are here

[dodoma_Ghana_walter (2).zip](https://github.com/IDEMSInternational/R-Instat/files/15491267/dodoma_Ghana_walter.2.zip)

I realised that boxplots on the monthly data show the pattern of some percentage points, so here was my first set of boxplots

image

Almost fine.  But this isn't a graph to show outliers, so let's change the coef from 1.5 upwards, so the lines for to the ends of the data.

Here is the graph:

image

That's fine - except for the really dry months of Juneto September.  And that is with a coefficient of 1000!

That's because more than 3/4 of the years are dry, so the interquartile range is zero.  

a) When I started this issue I just had one solution.  I found that if you make coef=NULL then all is ok, and that is really easy to do as a small tweak of the script. 

image

Problem solved.  But

b) Let's think a bit more.  Bit silly to use boxplots on data with so many zeros, without taking account of them!  So filter the data for zero and then make the width proportional to sample size left.  So this would be better:

image

That's a better graph!  Interesting example for teaching too.  And the fact the last (variable width) graph is much more sensible the solution obvious.  We do0n't try and be clever and permit NULL.  It is so rarely needed.  It can be done, via the script, but shouldn't be needed, because there is a better way?

### PR DESCRIPTION
**Pull request by Vitalis95**
Fixes #8995 
@rdstern @N-thony , have look at it

### CODE REVIEW

**Review by rdstern — DISMISSED** [review]
Great @vitalis.  Looks fine.  @N-thony over to you.

**Comment by vitalis** [comment]
@rdstern 
The owner if the PR is @Vitalis95 and not me (@vitalis)

File: instat/dlgBoxPlot.vb [hunk 1]
```diff
@@ -833,6 +836,23 @@ Public Class dlgBoxplot
...
+                ucrInputWidth.Visible = ucrChkWidth.Checked
+            Else
+                ucrChkWidth.Visible = False
+                ucrInputWidth.Visible = False
+            End If
+        Else
+            ucrChkWidth.Visible = False
+            ucrInputWidth.Visible = False
+        End If
+    End Sub
```
[the remark this answers is not in the document]
Comment (Vitalis95) [reply instat/dlgBoxPlot.vb]: @N-thony , With this, ucrInputWidth is still visible when the datatype is not numeric. I have simplified the initial code

**Review by rdstern — APPROVED** [review]
@N-thony still seems ok to me.  Back to you

**Pull request merged**